Makefile

```diff
@@ -12,138 +12,258 @@
+	manifest_image_folder=`echo "$${registry_prefix}${IMAGE_TAG}" | sed "s|/|_|g" | sed "s/:/-/"`; \
+	for arch in $(ALL_ARCH.windows); do \
+		for osversion in $(ALL_OSVERSIONS.windows); do \
+			BASEIMAGE=mcr.microsoft.com/windows/nanoserver:$${osversion}; \
+			full_version=`docker manifest inspect $${BASEIMAGE} | jq -r '.manifests[0].platform["os.version"]'`; \
+			sed -i -r "s/(\"os\"\:\"windows\")/\0,\"os.version\":\"$${full_version}\"/" "${HOME}/.docker/manifests/$${manifest_image_folder}/$${manifest_image_folder}-windows-$${osversion}-$${arch}"; \
+		done; \
+	done
+	docker manifest push --purge $(IMAGE_TAG)
+	docker manifest inspect $(IMAGE_TAG)
+
+## --------------------------------------
+## E2E Testing
+## --------------------------------------
+.PHONY: e2e-bootstrap
+e2e-bootstrap: $(HELM) $(BATS) $(KIND) $(KUBECTL) $(ENVSUBST) #setup all required binaries and kind cluster for testing
+ifndef TEST_WINDOWS
+	$(MAKE) setup-kind
 endif
-	docker pull $(IMAGE_TAG) || make e2e-container
+	docker pull $(IMAGE_TAG) || $(MAKE) e2e-container
 
 .PHONY: setup-kind
-setup-kind:
-	# Download and install kind
-	kind --version | grep -q $(KIND_VERSION) || (curl -L https://github.com/kubernetes-sigs/kind/releases/download/v$(KIND_VERSION)/kind-linux-amd64 --output kind && chmod +x kind && mv kind /usr/local/bin/)
-	# Create kind cluster
-	kind delete cluster || true
+setup-kind: $(KIND)
+	# Create kind cluster if it doesn't exist
+	if [ $$(kind get clusters) ]; then kind delete cluster; fi
 	kind create cluster --image kindest/node:v$(KUBERNETES_VERSION)
 
 .PHONY: e2e-container
 e2e-container:
-	docker buildx rm container-builder || true
-	# only moby/buildkit:foreign-mediatype works on building Windows image now
-	# https://github.com/moby/buildkit/pull/1879
-	# Github issue: https://github.com/moby/buildkit/issues/1877
-	docker buildx create --use --name=container-builder --driver-opt image=moby/buildkit:v0.7.2
 ifdef TEST_WINDOWS
-		docker buildx build --no-cache --build-arg LDFLAGS=$(LDFLAGS) -t $(IMAGE_TAG)-linux-amd64 -f docker/Dockerfile --platform="linux/amd64" --push .
-		docker buildx build --no-cache --build-arg LDFLAGS=$(LDFLAGS) -t $(IMAGE_TAG)-windows-1809-amd64 -f docker/windows.Dockerfile --platform="windows/amd64" --push .
-		docker manifest create $(IMAGE_TAG) $(IMAGE_TAG)-linux-amd64 $(IMAGE_TAG)-windows-1809-amd64
-		docker manifest inspect $(IMAGE_TAG)
-		docker manifest push --purge $(IMAGE_TAG)
+	$(MAKE) container-all push-manifest
 else
-		REGISTRY="e2e" make image
-		kind load docker-image --name kind e2e/secrets-store-csi:$(IMAGE_VERSION)
+	$(MAKE) container
+	kind load docker-image --name kind $(IMAGE_TAG)
 endif
 
-.PHONY: install-helm
-install-helm:
-	helm version --short | grep -q v3 || (curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash)
+.PHONY: e2e-test
+e2e-test: e2e-bootstrap # run test for windows
```

Comment:
is this windows only tests?

Reply:
that's correct. The target is baked in kubetest: https://github.com/kubernetes/test-infra/blob/master/kubetest/aksengine.go#L1346-L1351